topiary/src/atom_collection.rs

```diff
@@ -327,7 +327,9 @@ impl AtomCollection {
             node.is_named()
         );
 
-        if node.child_count() == 0 || self.specified_leaf_nodes.contains(&node.id()) {
+        if node.end_byte() == node.start_byte() {
+            log::warn!("Skipping zero-byte node: {node:?}");
```

Comment:
I don't know if I would put `warn` here: in my opinion, `warn` should inform the user that something wrong (but not fatal) is happening with their input, or the query file, and should/could be fixed by them.
Here the error is entirely on topiary's side, the user cannot do anything about it. I would replace it by `info` or `debug`